numismatic/feeds/base.py

```diff
@@ -245,52 +249,33 @@ class WebsocketClient(abc.ABC):
 
     exchange = None
     websocket_url = None
-    _websocket_map = dict()
-    _semaphore_map = dict()
-
-    @property
-    def websocket(self):
-        try:
-            return self._websocket_map[self.websocket_url]
-        except KeyError:
-            return None
-    
-    @property
-    def semaphore(self):
-        if self.websocket_url is None:
-            return None
-        try:
-            return WebsocketClient._semaphore_map[self.websocket_url]
-        except KeyError:
-            semaphore = asyncio.BoundedSemaphore(1)
-            WebsocketClient._semaphore_map[self.websocket_url] = semaphore
-            return semaphore
-        except Exception as ex:
-            logger.error(ex)
-            logger.error(packet)
-            raise
-
-    async def _connect(self):
+    _websocket_future = None
+    __listener = None
+    subscriptions = []
+            
+    def __attrs_post_init__(self):
+        asyncio.ensure_future(self._listener())
+
+    async def _connect(self, websocket_url=None):
         '''
-            Connects to websocket. Uses a semaphore
-            to ensure that only one connection at
-            a time will happen
+            Connects to websocket. Uses a future to ensure that only one
+            connection at a time will happen
         '''
-        with (await self.semaphore):
```

Comment:
As said previously, I didn't like introducing semaphores because it introduces another concept. Asyncio documentation doesn't usually talk about them. Async programmers are more familiar with Futures and Tasks so I prefer using those concepts to solve the problem here.